chartfx-math/src/main/java/de/gsi/math/MathBaseGenBase.java

```diff
@@ -0,0 +1,428 @@
+
+    public static double cosH(double arg0) {
+        return java.lang.Math.cosh(arg0);
+    }
+
+    public static boolean even(long a) {
+        return !((a & 1) != 0);
+    }
+
+    public static double exp(double x) {
+        return java.lang.Math.exp(x);
+    }
+
+    public static boolean finite(double x) {
+        return !(Double.isInfinite(x) || Double.isNaN(x));
+    }
+
+    public static double floor(double x) {
+        return java.lang.Math.floor(x);
+    }
+
+    public static int floorNint(double x) {
+        return nInt(java.lang.Math.floor(x));
+    }
+
+    public static boolean isNaN(double x) {
+        return Double.isNaN(x);
+    }
+
+    public static double ldExp(double x, int exp) {
+        return x * java.lang.Math.pow(2, exp);
+    }
+
+    public static double log2(double x) {
+        return java.lang.Math.log(x) / LOG_2;
+    }
+
+    //// codegen: double -> float, int, long, short
+    public static double max(double a, double b) {
+        return a >= b ? a : b;
+    }
+    //// end codegen
+
+    //// codegen: double -> float, int, long, short
+    public static double min(double a, double b) {
+        return a <= b ? a : b;
+    }
+    //// end codegen
+
+    /**
+     * Return next prime number after x, unless x is a prime in which case x is returned.
+     * 
+     * @param x input
+     * @return next prime number greater/equal x
+     */
+    public static long nextPrime(long x) {
+        if (x <= 2)
+            return 2;
+        if (x == 3)
+            return 3;
+        if (x % 2 == 0)
+            x++;
+
+        long sqr = (long) java.lang.Math.sqrt(x) + 1;
+
+        for (;;) {
+            long n;
+            for (n = 3; (n <= sqr) && ((x % n) != 0); n += 2)
+                ;
+            if (n > sqr) {
+                return x;
+            }
+            x += 2;
+        }
+    }
+
+    /**
+     * @param x input
+     * @return Round to nearest integer. Rounds half integers to the nearest even integer.
+     */
+    public static int nInt(double x) {
+        int i;
+        if (x >= 0) {
+            i = (int) (x + 0.5);
+            if ((x + 0.5 == i) && ((i & 1) > 0))
+                i--;
+        } else {
+            i = (int) (x - 0.5);
+            if ((x - 0.5 == i) && ((i & 1) > 0))
+                i++;
+        }
+        return i;
+    }
+
+    /**
+     * @param x input
+     * @return Round to nearest integer. Rounds half integers to the nearest
+     */
+    public static int nInt(float x) {
+        int i;
+        if (x >= 0) {
+            i = (int) (x + 0.5f);
+            if ((x + 0.5f == i) && ((i & 1) > 0))
+                i--;
+        } else {
+            i = (int) (x - 0.5);
+            if ((x - 0.5f == i) && ((i & 1) > 0))
+                i++;
+        }
+        return i;
+    }
+
+    public static boolean odd(long a) {
+        return (a & 1) != 0;
+    }
+
+    public static double pow(double x, double y) {
+        return java.lang.Math.pow(x, y);
```

Comment:
Codacy found an issue: [Unnecessary use of fully qualified name 'java.lang.Math.pow' due to existing implicit import 'java.lang.*'](https://app.codacy.com/manual/GSI/chart-fx/pullRequest?prid=5780327)